step/src/main/java/com/google/sps/servlets/InfoServlet.java

```diff
@@ -53,7 +53,12 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     String email = user.getEmail();
     String id = user.getUserId();
     
-    DatabaseHandler.addUser(firstName, lastName, dayBirth, monthBirth, yearBirth, email, id);
+    // Ensuring that only users who have filled all of the info fields out are added to the database
+    if (firstName != null && lastName != null && dayBirth != 0 && monthBirth != 0 &&
+      yearBirth != 0 && email != null && id != null) {
```

Comment:
Wouldn't this cause a problem when someone's yearBirth is 2000 (the preference form only asks for the last 2 digits, so 00) and that user would never be added to the database?

Reply:
Ah, I see you changed yearBirth to be 4 digits on the form but can you verify that the user actually entered 4 digits? Maybe call string.length() before casting to an int in this servlet?